src/Elastic.Apm/Model/Payload/Transaction.cs

```diff
@@ -14,23 +13,28 @@ namespace Elastic.Apm.Model.Payload
 {
 	internal class Transaction : ITransaction
 	{
-		internal readonly DateTimeOffset Start;
-
 		private readonly Lazy<Context> _context = new Lazy<Context>();
 		private readonly ScopedLogger _logger;
 		private readonly IPayloadSender _sender;
 
+		private readonly DateTimeOffset _start;
+
 		public Transaction(IApmAgent agent, string name, string type)
 			: this(agent.Logger, name, type, agent.PayloadSender) { }
 
 		public Transaction(IApmLogger logger, string name, string type, IPayloadSender sender)
 		{
 			_logger = logger?.Scoped(nameof(Transaction));
 			_sender = sender;
-			Start = DateTimeOffset.UtcNow;
+			_start = DateTimeOffset.Now;
+
 			Name = name;
 			Type = type;
-			Id = Guid.NewGuid();
+			var rnd = new Random();
```

Comment:
Hi @gregkalapos ,
Are you sure that using random is a good solution? https://stackoverflow.com/questions/10741433/c-sharp-random-function-returning-the-same-value 
For our projects, when we need random numbers we prefer to use snowflake alg https://github.com/RobThree/IdGen

Reply:
We have to generate 64 and 128bit random ids. That lib seems to only generate 64bit ids. Or did I miss something? I moved this code to a central place into a static instance, so we won't have any collisions. Probably we have room for optimisations here, but for now I think we are good here.

Reply:
Hi @gregkalapos , 
You already changed that code, previously you used just new Random()